pennylane/_device.py

```diff
@@ -322,6 +322,24 @@ def execute(self, queue, observables, parameters={}, **kwargs):
 
             return self._asarray(results)
 
+    def batch_execute(self, circuits):
+        """Execute a batch of quantum circuits on the device.
+
+        The circuits are represented by tapes, and they are executed one-by-one using the
+        tape's `execute` method. This ensures that the correct interface is applied to the tape.
+        The results are collected in a list.
+
+        For plugin developers: This function should be overwritten if the device can efficiently run multiple
+        circuits on a backend, for example using parallel and/or asynchronous executions.
+
+        Args:
+            circuits (list[~.tapes.QuantumTape]): circuits to execute on the device
+
+        Returns:
+            list[array[float]]: list of measured value(s)
+        """
+        return [circuit.execute(self) for circuit in circuits]
```

Comment:
The debate between `circuit.execute(dev)` and `device.execute(circuit)` reminds me of the perenial debate in Python over `str.join(iterable)` vs `iterable.join(str)` 😆

Reply:
Although, in our case it's not fully semantics, more so that the class with the `execute` method must be the one with the interface (and in this case, the tapes have the interface).

Reply:
Actually, I'm really confused now why we're doing `circuit.execute()` and not `device.execute()` :laughing: (sorry to replicate a previous discussion!) I thought we are on the device side and can simply loop over `device.execute()`?

Reply:
Yes, that's what I thought and wrote in a previous version of the PR.

It's interface magic: `circuit.execute(self)` is *not* the same as `device.execute()`. The tape's execute function is wrapped by a whole lot of tape logic. 

Last night I realised this is not so trivial after all, since devices that overwrite this function need to somehow get access to the interfaces. @josh, I think we might need a `tape.batch_execute` function after all, which calls the `dev.batch_execute` function, so we can here revert to `device.execute(circuit)` and the device does not need to worry about interfaces?

Reply:
Yep: the main difference is that

* `tape.execute()` knows how to unwrap ML tensors and convert them to NumPy arrays
* `tape.execute()` is a primitive for the ML framework; there will be a backwards method associated with it.

In comparison, `device.execute()` is pretty low level, doesn't know how to unwrap tensors, and doesn't provide a backward gradient method.

Reply:
If `device.batch_execute()` gets a list of tapes with no interface (as it currently does from the gradient methods), it should work fine as is. Or is it causing errors?

Further down the line, we may allow users to batch execute tapes _with_ interfaces. We will also have to grapple with how to combine and return the jacobian information for each individual tape; this is the point where some tape-level `batch_execute` makes sense, as it will require interfaces.